### PR TITLE
Add support for google cloud

### PR DESCRIPTION
**Pull request by mattlindsey**
Modifications to allow running the site on docker locally and on Google Cloud.

It's currently dying on trying to connect to Google Cloud SQL postgres with this:
  PG::ConnectionBad: could not connect to server: No such file or directory
  
It seems to happen first when 'rake db:setup' tries to create the db and do migrations.  And the postgres logs don't show it hitting the database.  And the 'websiteone-production' db doesn't get created unless I do it manually.

I have checked a lot of things, and it may be a problem with the encrypted db password using the 'master key' method in rails, but I also seem to get the error just defining the password in the cloud service.  Possibly the master key or 'rails secret key' didn't get saved correctly.

### CODE REVIEW

**Comment by mtc2013** [comment]
have a look at this for sql issues

https://dev.to/ku6ryo/access-cloud-sql-from-docker-on-cloud-run-41b6

**Comment by mattlindsey** [comment]
Also this for postgres stuff needed
https://dev.to/ku6ryo/access-cloud-sql-from-docker-on-cloud-run-41b6

**Comment by mattlindsey** [comment]
I think this looks good now.
Turns out this doc was the latest/best way to do it, rather than the older 4 step tutorial which is more complicated:
https://cloud.google.com/ruby/rails/run

File: .gitignore [hunk 1]
```diff
@@ -102,4 +102,5 @@ certbot.log
 rerun.txt
 
 # Ignore VCR
-features/support/fixtures/cassettes/**
\ No newline at end of file
+features/support/fixtures/cassettes/**
```
Comment (mtc2013) [inline .gitignore:105]: is it necessary to put this re-enabling of vcr in this pr?

File: Gemfile [hunk 2]
```diff
@@ -36,6 +36,7 @@ gem 'faker'
 gem 'font-awesome-rails'
 gem 'friendly_id'
 gem 'geocoder'
+gem 'google-cloud-storage'
```
Comment (mtc2013) [inline Gemfile]: do we need this in this pr?

File: config/application.rb [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 require_relative 'boot'
 require 'rails'
+require "active_storage/engine"
```
Comment (mtc2013) [inline config/application.rb]: do we end up needing this?

File: config/storage.yml [hunk 4]
```diff
@@ -15,11 +15,12 @@ local:
 #   bucket: your_own_bucket
 
 # Remember not to checkin your GCS keyfile to a repository
-# google:
-#   service: GCS
-#   project: your_project
-#   credentials: <%= Rails.root.join("path/to/gcs.keyfile") %>
-#   bucket: your_own_bucket
+# [START cloudrun_rails_storage]
+google:
```
Comment (mtc2013) [inline config/storage.yml]: do we need this in here?

File: db/migrate/20221221225406_create_active_storage_tables.active_storage.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# This migration comes from active_storage (originally 20170806125915)
+class CreateActiveStorageTables < ActiveRecord::Migration[5.2]
```
Comment (mtc2013) [inline db/migrate/20221221225406_create_active_storage_tables.active_storage.rb]: maybe we don't need this in this pr?

File: Dockerfile [hunk 6]
```diff
@@ -1,41 +1,66 @@
...
 
-RUN curl -sL https://deb.nodesource.com/setup_16.x | bash -
-RUN apt-get update -qq && apt-get install -y build-essential \
-    libpq-dev nodejs  \
-    libqt5webkit5-dev dos2unix \
-    gstreamer1.0-plugins-base gstreamer1.0-tools gstreamer1.0-x
+# [START cloudrun_rails_base_image]
+# Pinning the OS to buster because the nodejs install script is buster-specific.
+# Be sure to update the nodejs install command if the base image OS is updated.
+FROM ruby:3.0-buster as base
```
Comment (mtc2013) [inline Dockerfile:7]: are we running 3.0.5 here with this?

File: .gitignore (hunk 1 above)
Comment (mattlindsey) [reply .gitignore:105]: Not necessary for google cloud, but leaving them disabled can cause confusion in development (as it did with me).  I don't think the PR that took them out gave any reason for taking it out, so I'd rather keep it the way it has always been unless there's a good reason.

File: Gemfile (hunk 2 above)
Comment (mattlindsey) [reply Gemfile]: I'll try taking all of the cloud storage stuff out.  There's a chance that it's necessary for asset pipeline, but hopefully not.

File: Dockerfile (hunk 6 above)
Comment (mattlindsey) [reply Dockerfile:7]: Yes.  I think they keep 'buster' up with the 'stable' version of ruby 3.0.x

**Comment by mattlindsey** [comment]
fixes #3819 
fixes #3813

**Comment by mattlindsey** [comment]
Based on the PR review comments, I tried to take out the Google Storage stuff but it seems that at least some of it is necessary, since we go back to getting this error:

Step #2 - “apply migrations”: PG::ConnectionBad: could not connect to server: No such file or directory
Step #2 - “apply migrations”: 	Is the server running locally and accepting
Step #2 - “apply migrations”: 	connections on Unix domain socket “/var/run/postgresql/.s.PGSQL.5432"?

(Also I can't figure out how to run 'rake db:rollback' on the migration on Google Cloud SQL, but I don't think that's the problem.)

I'll try a few things things ...

File: .gitignore (hunk 1 above)
Comment (mattlindsey) [reply .gitignore:105]: actually looks like I just accidentally removed the last blank line, which doesn't seem to hurt anything

File: Gemfile (hunk 2 above)
Comment (mattlindsey) [reply Gemfile]: I took it out.

File: config/application.rb (hunk 3 above)
Comment (mattlindsey) [reply config/application.rb]: I took it out.

File: config/storage.yml (hunk 4 above)
Comment (mattlindsey) [reply config/storage.yml]: I took it out.

File: db/migrate/20221221225406_create_active_storage_tables.active_storage.rb (hunk 5 above)
Comment (mattlindsey) [reply db/migrate/20221221225406_create_active_storage_tables.active_storage.rb]: I took it out.

**Comment by mattlindsey** [comment]
@mtc2013.  See it this looks good now.  We should squash the commits when merging since there's so many.

**Review by mtc2013 — APPROVED** [review]
Let's pull this in and open a ticket for streamlining the deploy to google cloud. It's not a showstopper since we're not doing continuous delivery right now, nor do we have a production site.

**Pull request merged**